Sol.java

```diff
@@ -0,0 +1,104 @@
+import java.util.*;
+
+public class Sol {
+    int[] in ;
+    HashMap < Integer, Integer > exp;
+    LinkedList < Move > moves_coll;
+    int zeropos;
+
+    public LinkedList<Move> move_cars(int[] in , int[] out) {
+        initialise( in , out);
+        solve();
+      
+        return moves_coll;
+    }
+
+    private void initialise(int[] in , int[] out) {
+        exp = new HashMap < Integer, Integer > ();
+        moves_coll = new LinkedList < Move > ();
+        this.in = in ;
```

Comment:
I wasn't referring to this algorithm in particular. I was saying that for an algorithm X that has a time complexity of O log n, creating a copy would render its complexity to O n